WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListViewModel.kt

```diff
@@ -81,16 +89,34 @@ class NotificationsListViewModel @Inject constructor(
         appPrefsWrapper.notificationPermissionsWarningDismissed = false
     }
 
-    fun markNoteAsRead(context: Context, notes: List<Note>) {
+    fun markNoteAsRead(context: Context, notes: List<Note>) = launch {
+        if (networkUtilsWrapper.isNetworkAvailable().not()) {
+            withContext(mainDispatcher) {
+                toastUtilsWrapper.showToast(R.string.error_network_connection)
+            }
+            return@launch
+        }
         notes.filter { it.isUnread }
             .map {
                 gcmMessageHandler.removeNotificationWithNoteIdFromSystemBar(context, it.id)
-                notificationsActionsWrapper.markNoteAsRead(it)
-                it.setRead()
-                it
-            }.takeIf { it.isNotEmpty() }?.let {
-                notificationsTableWrapper.saveNotes(it, false)
+                it.apply { setRead() }
+            }.takeIf { it.isNotEmpty() }?.let { notes ->
+                // update the UI before the API request
+                notificationsTableWrapper.saveNotes(notes, false)
                 eventBusWrapper.post(NotificationsChanged())
+                // mark notes as read, this might wait for a long time
+                notificationsActionsWrapper.markNoteAsRead(notes)?.let { result ->
```

Comment:
In my case, it spent 8.2s on making the API request (22 unread notes). It looks weird when we mark all as read, but it update the UI after a long time. That's why I want to update the UI before the API request.

Reply:
Thank you for the clarification Jarvis 🙇